py/objdict.c

```diff
@@ -643,7 +644,7 @@ STATIC mp_obj_t dict_move_to_end(size_t n_args, const mp_obj_t *pos_args, mp_map
     enum { ARG_self, ARG_key, ARG_last };
     static const mp_arg_t allowed_args[] = {
         { MP_QSTR_, MP_ARG_REQUIRED | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE } },
```

Comment:
This is not generally how we do `self` handling. Instead, delete this line, and use the suggestions below.
```suggestion
```

Reply:
Change the `mp_arg_parse_all()` to:
```c
    mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
```
Change the `self` assignment to:
```c
    mp_obj_dict_t *self = MP_OBJ_TO_PTR(pos_args[0]);
```